### PR TITLE
feat: Introduce Intermediate Representation

### PR DESCRIPTION
**Pull request by raulb**
fixes: https://github.com/meroxa/product/issues/496.

Screenshot from 2022-09-01 12-42-53

### CODE REVIEW

File: platform/v2/resources.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	Name        string
+	Type        string
+	Source      bool
+	Destination bool
+	Collection  string
+	client      meroxa.Client
+	v           *Turbine
+}
+
+// TODO: Check if it's still valid
```
Comment (janelletavares) [inline platform/v2/resources.go:21]: i believe this is used for feature branch support

File: local/local.go [hunk 2]
```diff
@@ -101,7 +100,7 @@ type fixtureRecord struct {
 }
 
 func readFixtures(path, collection string) (turbine.Records, error) {
-	b, err := ioutil.ReadFile(path)
+	b, err := os.ReadFile(path)
```
Comment (janelletavares) [inline local/local.go:103]: lint

File: local/local.go [hunk 3]
```diff
@@ -146,3 +145,7 @@ func (t Turbine) RegisterSecret(name string) error {
 
 	return nil
 }
+
+func (t Turbine) HandleSpec() (string, error) {
```
Comment (raulb) [inline local/local.go:149]: I don't think we need to implement this for local.

**Comment by raulb** [comment]
@janelletavares this looks good to go. I only have a question on `HandleSpec` for `local`.

File: platform/platform.go [hunk 4]
```diff
@@ -52,11 +52,13 @@ func New(deploy bool, imageName, appName, gitSha string) *Turbine {
 	}
 }
 
+// TODO: Remove once everything is under IR
```
Comment (jayjayjpg) [inline platform/platform.go:55]: Thank you for all the comments!

File: platform/v2/functions_test.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+)
+
+type testFunction struct {
+}
+
+func (testFunction) Process(r []turbine.Record) []turbine.Record {
+	return r
+}
+
+func TestGetFunction(t *testing.T) {
```
Comment (jayjayjpg) [inline platform/v2/functions_test.go:18]: Nice test coverage! :sparkles:

File: platform/v2/resources_test.go [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+				resources: tc.resources,
+			}
+			list, err := testTurbine.ListResources()
+			require.NoError(t, err)
+			require.Equal(t, list, tc.output)
+		})
+	}
+}
+
+func TestRecords(t *testing.T) {
```
Comment (jayjayjpg) [inline platform/v2/resources_test.go:63]: :sparkles: :sparkles: :sparkles:

**Review by jayjayjpg — APPROVED** [review]
This reads great, thank you for the swift addition and all of the test coverage! :sparkles:

File: local/local.go (hunk 3 above)
Comment (janelletavares) [reply local/local.go:149]: HandleSpec is part of the TurbineRunner interface, so it thought local was part of that too, but apparently not.  maybe i'll rename it to TurbinePlatformRunner for clarity. and then I'll remove this.  Thanks! Also, I assumed we'd immediately rename this function when we send the POST deployment request.

**Pull request merged**